pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotTenantRestletResource.java

```diff
@@ -196,6 +198,42 @@ public String listInstanceOrToggleTenantState(
     }
   }
 
+  /**
+   * This method expects a tenant name and will return a list of tables tagged on that tenant. It assumes that the
+   * tagname is for server tenants only. This method will only read CURRENTSTATES in zk.
+   * @param tenantName
+   * @return
+   */
+  @GET
+  @Path("/tenants/{tenantName}/tables")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "List tables on a a server tenant")
+  @ApiResponses(value = {
+      @ApiResponse(code = 200, message = "Success"),
+      @ApiResponse(code = 500, message = "Error reading list")
+  })
+  public String getTablesOnTenant(
+      @ApiParam(value = "Tenant name", required = true) @PathParam("tenantName") String tenantName) {
+    return getTablesServedFromTenant(tenantName);
+  }
+
+  private String getTablesServedFromTenant(String tenantName)
+      throws JSONException {
+    Set<String> tables = new HashSet<String>();
+    JSONObject resourceGetRet = new JSONObject();
+
+    for (String table : pinotHelixResourceManager.getAllTables()) {
+      TableConfig tableConfig = pinotHelixResourceManager.getTableConfig(table);
+      String tableConfigTenant = tableConfig.getTenantConfig().getServer();
+      if (tenantName.equals(tableConfigTenant)) {
+        tables.add(table);
```

Comment:
The logic here shows if there's no segments for this table, this table can still be returned.

Reply:
I think that's the correct behavior.